api/config/v1alpha1/metric_types.go

```diff
@@ -10,6 +10,14 @@ type ProxyMetrics struct {
 	Prometheus *PrometheusProvider `json:"prometheus,omitempty"`
 	// Sinks defines the metric sinks where metrics are sent to.
 	Sinks []MetricSink `json:"sinks,omitempty"`
+	// Matches defines configuration for selecting specific metrics instead of generating all metrics stats
+	// that are enabled by default. This helps reduce CPU and memory overhead in Envoy.
+	// When Matches is nil(not set), gateway by default create and expose only a subset of
```

Comment:
lines 15-19 can be removed, imo
also because its hard to define in docs what exactly the default stats are, for e.g. we currently also output `http` stats